src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -941,5 +941,12 @@ private CorInfoHelpFunc getNewArrHelper(CORINFO_CLASS_STRUCT_* arrayCls)
 
             return CorInfoHelpFunc.CORINFO_HELP_NEWARR_1_VC;
         }
+
+        private void* getFieldAddress(CORINFO_FIELD_STRUCT_* field, ref void* ppIndirection)
```

Comment:
Instead of duplicating this, maybe just make `GetFieldRvaData` virtual on `Compilation` and override it for CoreCLR?

Eventually, I would like us to end up with a more abstract `Compilation` class that makes sense for both R2R and CoreRT, and have a `CoreRTCompilation` and `ReadyToRunCompilation` deriving from it. This will keep us on a path to reach that goal. (I don't like we currently have methods on `Compilation` that don't work for R2R.)

Reply:
Jit has no visibility into ISymbolNode. The check needs to be in this method. Something like

```csharp
ISymbolNode s = _compilation.GetFieldRvaData(fieldDesc);
if (s.RepresentsIndirectionCell)
{
    ppIndirection = ObjectToHandle(s);
    return null;
}

return ObjectToHandle(s)
```

Reply:
Fixed in 3rd commit, thanks for your clarification.